src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -356,16 +364,26 @@ public static CodeAction Create(string title, ImmutableArray<CodeAction> nestedA
             return new CodeActionWithNestedActions(title, nestedActions, isInlinable);
         }
 
+        /// <summary>
+        /// Creates a code action provider name.
+        /// </summary>
+        /// <param name="provider">Specific Code Action Provider (<see cref="CodeRefactoringProvider"/> or <see cref="CodeRefactoringProvider"/> implementation) of the <see cref="CodeAction"/> group.</param>
+        /// <param name="implementation">Specific Code Action implementation from the Code Action Provider.</param>
+        public static string CreateProviderName(string provider, string implementation)
```

Comment:
Needs to be added to published API.

Reply:
If the ProviderName is just made up of PredefinedCodeFixProviderNames and PredefinedCodeRefactoringProviderNames, we could avoid needing this method here.